python/ray/autoscaler/v2/instance_manager/config.py

```diff
@@ -18,129 +29,207 @@ class Provider(Enum):
 
 
 class IConfigReader(ABC):
-    """An interface for reading cluster config."""
+    """An interface for reading Autoscaling config.
+
+    A utility class that converts the raw autoscaling configs into
+    various data structures that are used by the autoscaler.
+    """
 
     @abstractmethod
-    def get_cluster_config(self) -> ClusterConfig:
-        """Returns the cluster config."""
+    def get_autoscaling_config(self) -> "AutoscalingConfig":
+        """Returns the autoscaling config."""
         pass
 
 
-class NodeProviderConfig(object):
+@dataclass
+class NodeTypeConfig:
     """
-    NodeProviderConfig is the helper class to provide instance
+    NodeTypeConfig is the helper class to provide node type specific configs.
+    This maps to subset of the `available_node_types` field in the
+    autoscaling config.
+    """
+
+    # Node type name
+    name: NodeType
+    # The minimal number of workers to be launched for this node type.
+    min_workers: int
```

Comment:
this has been renamed to `min_worker_nodes`

Reply:
oh wasn't aware this was already merged. but yeah, good call!

Reply:
Ah sorry it actually hasn't been merged yet, we were waiting until after 2.9.  I can update it in my PR

Reply:
sounds good - i will do the renaming for the dst fields, but leave it to you for the config field name.